sdk/core/azure-core/src/main/java/com/azure/core/util/SharedExecutorService.java

```diff
@@ -0,0 +1,350 @@
+ *     times the number of available processors.</li>
+ *     <li>{@code azure.sdk.sharedpool.keepalivemillis} - The keep alive time for threads in the shared executor
+ *     service. If not set, it defaults to 60 seconds.</li>
+ *     <li>{@code azure.sdk.sharedpool.virtual} - A boolean flag to indicate if the shared executor service should use
+ *     virtual threads. If not set, it defaults to true. Ignored if virtual threads are not available in the
+ *     runtime.</li>
+ * </ul>
+ *
+ * If a custom executor service is set using {@link #setExecutorService(ExecutorService)}, it updates all references to
+ * the shared executor service. Meaning, if another area in code already had a reference to the shared instance, it will
+ * now use the passed executor service to execute tasks.
+ * <p>
+ * Calls to {@link #shutdown()} and {@link #shutdownNow()} are not supported. If the default shared executor service is
+ * being used, it is bound to the lifecycle of the application and will be shutdown when the application is shutdown.
+ * If a custom executor service is set using {@link #setExecutorService(ExecutorService)}, it is the responsibility of
+ * the caller to manage the lifecycle of the executor service.
+ * <p>
+ * If a custom executor service is set using {@link #setExecutorService(ExecutorService)}, and it is shutdown or
+ * terminated, the shared executor service will be reset to the default shared executor service.
+ */
+@SuppressWarnings({ "resource" })
+public final class SharedExecutorService implements ExecutorService {
+    private static final ClientLogger LOGGER = new ClientLogger(SharedExecutorService.class);
+
+    private static final SharedExecutorService INSTANCE = new SharedExecutorService();
+
+    // Shared thread counter for all instances of SharedExecutorService created using the empty factory method.
+    private static final AtomicLong AZURE_SDK_THREAD_COUNTER = new AtomicLong();
+    private static final String AZURE_SDK_THREAD_NAME = "azure-sdk-global-thread-";
+
+    // The thread pool size for the shared executor service.
+    //
+    // This uses the configuration setting 'azure.sdk.sharedpool.size' if set, otherwise it defaults to 10 times the
+    // number of available processors.
+    // If 'azure.sdk.sharedpool.size' is set to a non-integer, negative value, or zero, the default value is used.
+    private static final int THREAD_POOL_SIZE
+        = Configuration.getGlobalConfiguration().get("azure.sdk.sharedpool.size", config -> {
+            try {
+                int size = Integer.parseInt(config);
+                if (size <= 0) {
+                    return 10 * Runtime.getRuntime().availableProcessors();
+                } else {
+                    return size;
+                }
+            } catch (NumberFormatException ignored) {
+                return 10 * Runtime.getRuntime().availableProcessors();
```

Comment:
maybe log an exception? If I misconfigured something, i'd want to know.

Also, if you used `ConfigurationProperty`, you could set default value, reuse int parsing and logging ;)

Reply:
Looked into using `ConfigurationProperty` but it had the downside that if the converter fails it doesn't fallback to the default value but throws the `NumberFormatException`. I'll add logging without usage of `ConfigurationProperty` for now as this is an implementation detail.

Reply:
I would be ok with changing config properties behavior. They are relatively new and we never advertised them, so it's very unlikely to break anyone. But I'm fine either way.